### ISSUE TITLE
cryptonite-0.25 test suite fails with "too many pending signals"

### ISSUE THREAD

**Issue by peti** [haskell-crypto/cryptonite#260]
Citing from https://hydra.nixos.org/build/85709602:

~~~
Running 1 test suites...
Test suite test-cryptonite: RUNNING...
cryptonite
  number
    num-bits:                            OK
      +++ OK, passed 100 tests.
    num-bits2:                           OK
      +++ OK, passed 100 tests.
    generate-param:                      OK (0.01s)
      +++ OK, passed 100 tests.
    generate-param2:                     OK
      +++ OK, passed 100 tests.
    generate-param-odd:                  OK
      +++ OK, passed 100 tests.
    generate-range:                      OK
      +++ OK, passed 100 tests.
    generate-prime:                      OK (0.67s)
      +++ OK, passed 100 tests.
    generate-safe-prime:                 OK (10.58s)
      +++ OK, passed 100 tests.
    marshalling:                         OK
      +++ OK, passed 100 tests.
    marshalling-kat-to-bytearray
      0:                                 OK
      1:                                 OK
      2:                                 OK
      3:                                 OK
      4:                                 OK
    marshalling-kat-to-integer
      0:                                 OK
      1:                                 OK
      2:                                 OK
      3:                                 OK
      4:                                 OK
  number.F2m
    addF2m
      commutative:                       OK
        +++ OK, passed 100 tests.
      associative:                       OK
        +++ OK, passed 100 tests.
      0 is neutral:                      OK
        +++ OK, passed 100 tests.
      nullable:                          OK
        +++ OK, passed 100 tests.
      works per bit:                     OK
        +++ OK, passed 100 tests.
    modF2m
      idempotent:                        OK
        +++ OK, passed 100 tests.
      upper bound:                       OK
        +++ OK, passed 100 tests.
      reach upper:                       OK
        +++ OK, passed 100 tests.
      lower bound:                       OK
        +++ OK, passed 100 tests.
      reach lower:                       OK
        +++ OK, passed 100 tests.
      additive:                          OK
        +++ OK, passed 100 tests.
    mulF2m
      commutative:                       OK
        +++ OK, passed 100 tests.
      associative:                       OK
        +++ OK, passed 100 tests.
      1 is neutral:                      OK
        +++ OK, passed 100 tests.
      0 is annihilator:                  OK
        +++ OK, passed 100 tests.
      distributive:                      OK
        +++ OK, passed 100 tests.
    squareF2m
      sqr(a) == a * a:                   OK
        +++ OK, passed 100 tests.
    invF2m
      1 / a * a == 1:                    OK
        +++ OK, passed 100 tests.
      1 / a == a (mod a^2-1):            OK
        +++ OK, passed 100 tests.
    divF2m
      1 / a == inv a:                    OK
        +++ OK, passed 100 tests.
      a / b == a * inv b:                OK
        +++ OK, passed 100 tests.
      a * b / b == a:                    OK
        +++ OK, passed 100 tests.
  hash
    KATs
      MD2
        1:                               OK
        2:                               OK
        3:                               OK
      MD4
        1:                               OK
        2:                               OK
        3:                               OK
      MD5
        1:                               OK
        2:                               OK
        3:                               OK
      SHA1
        1:                               OK
        2:                               OK
        3:                               OK
      SHA224
        1:                               OK
        2:                               OK
        3:                               OK
      SHA256
        1:                               OK
        2:                               OK
        3:                               OK
      SHA384
        1:                               OK
        2:                               OK
        3:                               OK
      SHA512
        1:                               OK
        2:                               OK
        3:                               OK
      SHA512/224
        1:                               OK
        2:                               OK
        3:                               OK
      SHA512/256
        1:                               OK
        2:                               OK
        3:                               OK
      RIPEMD160
        1:                               OK
        2:                               OK
        3:                               OK
      Tiger
        1:                               OK
        2:                               OK
        3:                               OK
      Skein256-256
        1:                               OK
        2:                               OK
        3:                               OK
      Skein512-384
        1:                               OK
        2:                               OK
        3:                               OK
      Skein512-512
        1:                               OK
        2:                               OK
        3:                               OK
      Whirlpool
        1:                               OK
        2:                               OK
        3:                               OK
      Keccak-224
        1:                               OK
        2:                               OK
        3:                               OK
      Keccak-256
        1:                               OK
        2:                               OK
        3:                               OK
      Keccak-384
        1:                               OK
        2:                               OK
        3:                               OK
      Keccak-512
        1:                               OK
        2:                               OK
        3:                               OK
      SHA3-224
        1:                               OK
        2:                               OK
        3:                               OK
      SHA3-256
        1:                               OK
        2:                               OK
        3:                               OK
      SHA3-384
        1:                               OK
        2:                               OK
        3:                               OK
      SHA3-512
        1:                               OK
        2:                               OK
        3:                               OK
      Blake2b-160
        1:                               test-cryptonite: too many pending signals
Test suite test-cryptonite: FAIL
~~~

**Comment by crockeea** [comment]
Tests pass for me on Windows 10 and Ubuntu 18.04.

**Comment by ahibal** [comment]
I have EXACTLY the same output when I run:
nix-env -f https://github.com/obsidiansystems/obelisk/archive/master.tar.gz -iA command

system:
debian 9
Kernel Linux 4.9.0-8-amd64 x86_64
2.9 GiB memory
AMD Athlon(tm) 64 X2 Dual Core Processor 6000+ × 2 

"too many pending signals" is it because I have low memory?

**Comment by ryantrinkle** [comment]
I'm seeing these tests take a very long time (more than an hour), but I'm not sure whether that's related.

**Comment by ocheron** [comment]
Can those having BLAKE2 failure try removing flag `support_aesni`?
I suspect SSSE3 is not supported on old AMD processors.

**Comment by pvgoran** [comment]
I have this exact problem with the current `master`. When I rebuild and test `cryptonite` with `--constraint="cryptonite -support_aesni"`, the problem goes away.

**Comment by pvgoran** [comment]
I re-posted this as #314, with more detailed description of the problem.

**Comment by ocheron** [comment]
Fixed in cryptonite-0.27 and now in Stackage Nightly.
There should not be error at runtime anymore with default flags, the cost is C compiler support being made more difficult for old versions.
